LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -348,4 +346,104 @@ public async Task GetLeaderboards()
         LeaderboardViewModel[] returned = await _apiClient.Get<LeaderboardViewModel[]>("/api/leaderboards", new());
         returned.Should().BeEquivalentTo(boards.Take(2), config => config.Excluding(lb => lb.Categories));
     }
+
+    [Test]
+    public async Task RestoreLeaderboard_OK()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Leaderboard deletedBoard = new()
+        {
+            Name = "Super Mario World",
+            Slug = "super-mario-world-to-restore",
+            DeletedAt = _clock.GetCurrentInstant()
+        };
+
+        context.Leaderboards.Add(deletedBoard);
+        await context.SaveChangesAsync();
+        deletedBoard.Id.Should().NotBe(default);
+
+        _clock.AdvanceMinutes(1);
+
+        LeaderboardViewModel res = await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/{deletedBoard.Id}/restore", new()
+        {
+            Jwt = _jwt
+        });
+
+        res.Id.Should().Be(deletedBoard.Id);
+        res.Slug.Should().Be(deletedBoard.Slug);
+        res.UpdatedAt.Should().Be(res.CreatedAt + Duration.FromMinutes(1));
+        res.DeletedAt.Should().BeNull();
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_Unauthenticated()
+    {
+        Func<Task<LeaderboardViewModel>> act = async () => await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/100/restore", new()
+        {
+            Jwt = ""
+        });
+
+        await act.Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Unauthorized);
+    }
+
+    [TestCase("restore-leaderboard-unauth1@example.com", "RestoreBoard1", UserRole.Confirmed)]
+    [TestCase("restore-leaderboard-unauth2@example.com", "RestoreBoard2", UserRole.Registered)]
+    public async Task RestoreLeaderboard_Unauthorized(string email, string username, UserRole role)
+    {
+        UserViewModel userModel = await _apiClient.RegisterUser(username, email, "P4ssword");
+
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        User? user = await context.Users.FindAsync([userModel.Id]);
+
+        context.Users.Update(user!);
+        user!.Role = role;
```

Comment:
Does the call to Update() do anything? When you fetch using FindAsync, it should start tracking changes to the returned object automatically. It might be simpler overall to just make this all one step by creating the user in the database directly instead of calling RegisterUser, fetching, updating, and then saving.

Reply:
Tried doing that but I keep getting "Invalid salt version". This is what I tried to do:
```cs
public async Task RestoreLeaderboard_Unauthorized(string email, string username, UserRole role)
{
    ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();

    context.Users.Add(new()
    {
        Email = email,
        Password = "P4ssword",
        Username = username,
        Role = role
    });

    await context.SaveChangesAsync();
```

Reply:
Yeah I know that. Just wanted to show that it wasn't possible for some reason (I know the user account tests work). And yeah; the FindAsync does track. Will update